frontend-site/src/components/Home/OfficerListing.vue

```diff
@@ -1,32 +1,65 @@
 <template>
   <v-layout row wrap class="officer-listing">
-    <v-flex
-      xs12 sm6 md4
-      v-for="(officer, i) in officers"
-      :key="i">
-      <officer-card :officer="officer"/>
+    <template v-if="!isLoading">
+      <v-flex
+        xs12 sm6 md4
+        v-for="(officer, i) in officers"
+        :key="i">
+        <officer-card :officer="officer"/>
+      </v-flex>
+      <v-flex v-if="officers.length === 0">
+        <v-card class="text-xs-center">
+          <v-card-text>No officer data found.</v-card-text>
+        </v-card>
+      </v-flex>
+    </template>
+    <v-flex v-else>
+      <v-card class="text-xs-center">
+          <v-card-text>Loading officer data...</v-card-text>
+        </v-card>
     </v-flex>
   </v-layout>
 </template>
 
 <script>
 import OfficerCard from '@/components/OfficerCard';
-import { mapState, mapActions } from 'vuex';
+import lugApi from '@/modules/LugApi';
 
 export default {
   components: {
     OfficerCard,
   },
   computed: {
-    ...mapState('officers', {
-      officers: 'data',
-    }),
-  },
-  methods: {
-    ...mapActions('officers', ['updateData']),
+    // TODO: factor out if other things need this
+    currentSemester () {
+      const currentDate = new Date();
+      const currentMonth = currentDate.getMonth() + 1; // getMonth is 0 indexed, so add 1
+      const currentYear = currentDate.getUTCFullYear();
+      let currentSemester = '';
+      if (currentMonth <= 5) { // up to May
+        currentSemester = 'SPRING';
+      } else if (currentMonth <= 8) { // up to August
+        currentSemester = 'SUMMER';
+      } else {
+        currentSemester = 'FALL';
+      }
+      return `${currentSemester}_${currentYear}`;
+    },
```

Comment:
I want to flag this up so we can standardize it between the frontend and the backend, so to confirm, we're saying that Spring = Jan-May, Summer = June-Aug, and Fall = Sept-Dec.

Reply:
Unless if there's some other way of figuring out the semester given a date, yes. I based it roughly off of the months for start/end listed in the [UIC Academic Calendar](https://catalog.uic.edu/ucat/academic-calendar/). I could make it more granular and compare actual start/end dates, but that would have to be done per academic year.